api/edgeproto/objs.go

```diff
@@ -455,6 +458,23 @@ func (s *Cloudlet) Validate(fmap objstore.FieldMap) error {
 	return nil
 }
 
+func (key *ZoneKey) ValidateKey() error {
+	if !util.ValidName(key.Organization) {
+		return fmt.Errorf("Invalid zone organization name %s", key.Organization)
+	}
+	if !util.ValidName(key.Name) {
+		return errors.New("Invalid zone name")
+	}
+	return nil
+}
+
+func (s *Zone) Validate(fmap objstore.FieldMap) error {
+	if err := s.GetKey().ValidateKey(); err != nil {
+		return err
+	}
+	return nil
+}
+
```

Comment:
I think if it's really boilerplate it would make sense to move it into the generator code. The validatekey function we could probably do by adding annotations to each key field for the validator function to use. Not sure if we could auto-gen the object validate function though.

Reply:
Yeah, it's either creating a validation function, or adding annotations in the proto file. It's a tradeoff I agree. Let me think about what possible things we can autogen, but I see how validation function is not as clear-cut  as I thought.